### ISSUE TITLE
select file at startup for staging/etc

### ISSUE THREAD

**Issue by teleshoes** [jesseduffield/lazygit#3788]
**Is your feature request related to a problem? Please describe.**
my primary use of lazygit is for line-by-line staging of single files, as i am editing them. i use a shortcut in vim to switch to lazygit, stage the lines, and commit them. it would be nice if the file i am currently editing was selected in lazygit.

**Describe the solution you'd like**
1) run: `lazygit --select-file src/file.cpp`
2) see lazygit start as normal
3) among files in the `Files` tab of panel[2], if that file's path (relative to the root of the repo) matches `src/file.cpp`, select that modified or unstaged file, as if selecting that row in `Files` with the kb or mouse (expanding dirs in the filetree as necessary)

**Describe alternatives you've considered**
- scripting keys at startup to do something like `xdotool key '/'`, `xdotool type $FILENAME`, `xdotool key Return` (this is awful, non-portable, and possibly subject to race conditions)
- creating a temporary custom cmd with yaml before startup that selects a file with a specific name (still have to press a button at startup)
- creating a permanent custom cmd that selects a file based on an environment variable $VIM_FILE and setting that (still have to press a button at startup)

**Additional context**
none

**Comment by mark2185** [comment]
As much as I love `lazygit`, for this specific thing maybe you would be better off with [vim-fugitive](https://github.com/tpope/vim-fugitive/) or maybe [vim-gitgutter](https://github.com/airblade/vim-gitgutter?tab=readme-ov-file#hunks) for staging current hunk. Just my two cents, consider it a possible alternative.

**Comment by teleshoes** [comment]
1) this is a generic feature, not vim-specific at all. i personally would use it from the cmdline about as much as i use it from vim. `lg path/to/file` is a lot easier than navigating an interactive Files panel.
2) vim-fugitive is nice, but it's staging tool is a pure wrapper around `git stage -p`, which does not support line-by-line staging.
3) vim-gitgutter is less universally endorsed so i haven't checked it out before, but it seems to claim that it supports line-by-line staging. i will look into it for sure, thanks!

**Comment by mark2185** [comment]
`vim-fugitive` does support partial staging, you just have to use visual line mode when staging.

I'll stop with the digression now.

**Comment by teleshoes** [comment]
> You can just use `/` to search.
yes, i did mention automating doing `/file` in my initial report


> `vim-fugitive` does support partial staging, you just have to use visual line mode when staging.
!!! did not know this, will try it out right now and see if it replaces lazygit for me. thanks!

**Comment by teleshoes** [comment]
hmm, line-by-line staging with `:Gdiff`+`:dp` works, but it seems that a unified diff like in git gui or lazygit is not feasible, since it is just wrapping vimdiff and writing the index from the resulting buffer.

a quick internet search did not turn up anyone getting functionality in vim similar to the staging ui from lazygit or git-gui, but maybe someone does have this? (all i want, really, is a terminal based, keyboard driven app that shows a nice, hunk-ed, unified diff of a single file against the index, AND allows easily staging a single +LINE or -LINE from within that diff.)

but like you say, this is a tangent, and lazygit could use this feature with or without vim

**Comment by mark2185** [comment]
This looks pretty close to `lazygit` to me:
![image](https://github.com/user-attachments/assets/77e36bfd-7cf0-4eb4-a08c-bf9cdf763998)

![image](https://github.com/user-attachments/assets/046f3d7e-ab30-4c14-b39c-8a3ee8e126e8)

In any case, this is a discussion better suited for the [`lazygit` discord channel](https://discord.gg/ehwFt2t4wt) since it is pretty off topic.
